### PR TITLE
Migrate AuditLogsSource to use generated reconciler.

### PR DESCRIPTION
**Pull request by nlopezgi**
4th PR part of # https://github.com/google/knative-gcp/issues/557

## Proposed Changes

- 🎁 Use reconciler gen to implement strongly typed reconcilers.
- Also fix calls that needed err.Error() in pubsub

### CODE REVIEW

File: pkg/reconciler/events/auditlogs/auditlogs.go [hunk 1]
```diff
@@ -61,121 +55,42 @@ type Reconciler struct {
...
 
-	return nil
+	return reconciler.NewEvent(corev1.EventTypeNormal, reconciledSuccessReason, `CloudAuditLogsSource reconciled: "%s/%s"`, s.Namespace, s.Name)
 }
 
 func (c *Reconciler) reconcileSink(ctx context.Context, s *v1alpha1.CloudAuditLogsSource) (string, error) {
 	sink, err := c.ensureSinkCreated(ctx, s)
 	if err != nil {
-		s.Status.MarkSinkNotReady("SinkCreateFailed", "failed to ensure creation of logging sink: %v", err)
+		s.Status.MarkSinkNotReady("SinkCreateFailed", "failed to ensure creation of logging sink: %v", err.Error())
```
Comment (nachocano) [inline pkg/reconciler/events/auditlogs/auditlogs.go]: `%s` if you do `err.Error()`, 
same below

**Comment by nachocano** [comment]
/lgtm
/approve
/cla: yes

the cosmetic comment can be addressed in the next PR

File: pkg/reconciler/events/auditlogs/auditlogs.go (hunk 1 above)
Comment (nachocano) [reply pkg/reconciler/events/auditlogs/auditlogs.go]: let's get this in so that we unblock @grac3gao ...
you can update in a follow up, no big deal

Comment (nachocano) [reply pkg/reconciler/events/auditlogs/auditlogs.go]: @grac3gao same here... would you mind addressing this nit in your bigger PR?

Comment (grac3gao) [reply pkg/reconciler/events/auditlogs/auditlogs.go]: Yeah, I'll change it.

**Pull request merged**